lib/gollum-lib/filter/tags.rb

```diff
@@ -241,7 +241,11 @@ def process_page_link_tag(link_part, pretty_name = nil)
   # Returns a Gollum::Page instance if a page is found, or nil otherwise
   def find_page_from_path(path)
     if Pathname.new(path).relative?
-      @markup.wiki.page(::File.join(@markup.dir, path))
+      page = @markup.wiki.page(::File.join(@markup.dir, path))
+      if page.nil? && @markup.wiki.link_compatibility # 4.x link compatibility option. Slow!
+        page = @markup.wiki.pages.detect {|page| page.path =~ /^(.*\/)?#{path}\..+/i}
```

Comment:
This is slow because it looks up all the wiki's pages first. A faster way of implementing this would have to make changes to `Page#path_match`.